src/java/htsjdk/samtools/cram/encoding/huffman/codec/HuffmanParamsCalculator.java

```diff
@@ -0,0 +1,153 @@
+package htsjdk.samtools.cram.encoding.huffman.codec;
+
+import htsjdk.samtools.cram.common.MutableInt;
+import htsjdk.samtools.cram.encoding.huffman.HuffmanCode;
+import htsjdk.samtools.cram.encoding.huffman.HuffmanTree;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * The purpose is to calculate Huffman encoding parameters.
+ * Critical: ALL values must be shown to the calculator before calling the calculate() method.
+ */
+public class HuffmanParamsCalculator {
+    private final HashMap<Integer, MutableInt> countMap = new HashMap<Integer, MutableInt>();
```

Comment:
You might want to consider using a library with type-specialized maps, e.g. [`trove4j`](http://trove.starlight-systems.com/). We've had some very bad experience with boxed maps applied to genomic data.